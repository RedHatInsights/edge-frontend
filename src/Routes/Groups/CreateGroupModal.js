import React from 'react';
import PropTypes from 'prop-types';
import validatorTypes from '@data-driven-forms/react-form-renderer/validator-types';
import componentTypes from '@data-driven-forms/react-form-renderer/component-types';
import Modal from '../../components/Modal';
import {
  createGroup,
  createInventoryGroup,
  addDevicesToGroup,
  validateGroupName,
  validateInventoryGroupName,
  addDevicesToInventoryGroup,
} from '../../api/groups';
import { nameValidator, useFeatureFlags } from '../../utils';
import apiWithToast from '../../utils/apiWithToast';
import { useDispatch } from 'react-redux';
import useInventoryGroups from '../../hooks/useInventoryGroups';
import { FEATURE_INVENTORY_WORKSPACES_RENAME } from '../../constants/features';

const asyncGroupNameValidation = async (value = '') => {
  // do not fire validation request for empty name
  if (value.length === 0) {
    return undefined;
  }
  const resp = await validateGroupName(value);
  if (resp.data.isValid) {
    // async validator has to throw error, not return it
    throw 'Group name already exists';
  }
};

const asyncInventoryGroupNameValidation = async (value = '') => {
  // do not fire validation request for empty name
  if (value.length === 0) {
    return undefined;
  }
  const resp = await validateInventoryGroupName(value);
  if (resp.results.length > 0 && resp.results[0].name === value) {
    // async validator has to throw error, not return it
    throw 'Group name already exists';
  }
};

const validatorMapper = {
  groupName: () => asyncGroupNameValidation,
};

const inventoryValidatorMapper = {
  groupName: () => asyncInventoryGroupNameValidation,
};

const createGroupSchema = (workspaceRename) => ({
  fields: [
    {
      component: componentTypes.TEXT_FIELD,
      name: 'name',
      label: `${workspaceRename ? 'Workspace' : 'Group'} name`,
      helperText:
        'Can only contain letters, numbers, spaces, hyphens ( - ), and underscores( _ ).',
      isRequired: true,
      autoFocus: true,
      validate: [
        // async validator has to be first in the list
        { type: 'groupName' },
        { type: validatorTypes.REQUIRED },
        { type: validatorTypes.MAX_LENGTH, threshold: 50 },
        nameValidator,
      ],
    },
  ],
});

const CreateGroupModal = ({
  isModalOpen,
  setIsModalOpen,
  deviceIds,
  reloadData,
}) => {
  const dispatch = useDispatch();

  const [inventoryGroupsEnabled] = useInventoryGroups(false);
  const useWorkspacesRename = useFeatureFlags(
    FEATURE_INVENTORY_WORKSPACES_RENAME
  );

  const handleCreateGroup = (values) => {
    const statusMessages = {
      onSuccess: {
        title: 'Success',
        description: `${values.name} has been created successfully`,
      },
      onError: {
        title: 'Error',
        description: `Failed to create ${
          inventoryGroupsEnabled && useWorkspacesRename ? 'workspace' : 'group'
        }`,
      },
    };

    let createGroupFunc;
    if (inventoryGroupsEnabled) {
      createGroupFunc = () => createInventoryGroup(values);
    } else {
      createGroupFunc = () => createGroup(values);
    }
    return apiWithToast(dispatch, createGroupFunc, statusMessages);
  };

  const handleAddDevicesToNewGroup = async (values) => {
    const statusMessages = {
      onSuccess: {
        title: 'Success',
        description: `System(s) have been added to ${values.name} successfully`,
      },
      onError: {
        title: 'Error',
        description: `Failed to add system to ${
          inventoryGroupsEnabled && useWorkspacesRename ? 'workspace' : 'group'
        }`,
      },
    };

    let addDevicesToGroupFunc;
    if (inventoryGroupsEnabled) {
      const { id } = await handleCreateGroup(values);
      addDevicesToGroupFunc = () => addDevicesToInventoryGroup(id, deviceIds);
    } else {
      const { ID } = await handleCreateGroup(values);
      addDevicesToGroupFunc = () => addDevicesToGroup(parseInt(ID), deviceIds);
    }
    apiWithToast(dispatch, addDevicesToGroupFunc, statusMessages);
  };

  return (
    <Modal
      isOpen={isModalOpen}
      closeModal={() => setIsModalOpen(false)}
      title={`Create ${
        inventoryGroupsEnabled && useWorkspacesRename ? 'workspace' : 'group'
      }`}
      submitLabel="Create"
      schema={createGroupSchema(inventoryGroupsEnabled && useWorkspacesRename)}
      onSubmit={deviceIds ? handleAddDevicesToNewGroup : handleCreateGroup}
      reloadData={reloadData}
      validatorMapper={
        inventoryGroupsEnabled ? inventoryValidatorMapper : validatorMapper
      }
    />
  );
};

export default CreateGroupModal;

CreateGroupModal.propTypes = {
  isModalOpen: PropTypes.bool,
  setIsModalOpen: PropTypes.func,
  reloadData: PropTypes.func,
  deviceIds: PropTypes.array,
};
