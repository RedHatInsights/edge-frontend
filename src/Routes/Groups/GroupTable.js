import React from 'react';
import PropTypes from 'prop-types';
import GeneralTable from '../../components/general-table/GeneralTable';
import { Link } from 'react-router-dom';
import { routes as paths } from '../../../package.json';
import { Tooltip } from '@patternfly/react-core';

const filters = [{ label: 'Name', type: 'text' }];

const columns = [
  { title: 'Name', type: 'name', sort: true },
  { title: 'Systems', type: 'name', sort: false },
  { title: 'Image', type: 'name', sort: false },
];

const GroupTable = ({
  data,
  isLoading,
  openModal,
  setIsModalOpen,
  handleRenameModal,
}) => {
  const actionResolver = (rowData) => {
    const { id, title } = rowData;
    return [
      {
        title: 'Rename',
        onClick: () => handleRenameModal(id, { name: title }),
      },
      {
        title: 'Delete',
        onClick: () => console.log('updating'),
      },
    ];
  };

  const buildRows = data.map((rowData) => {
    const { ID, Name, Devices } = rowData;
    const systems = Devices ?? [];

    // temp static data to show dif version of mockups
    const image = (
      <div>
        <Tooltip
          content={
            <div>
              <p>Golden Image</p>
              <p> Super Golden Image</p>
            </div>
          }
        >
          <span>Multiple images</span>
        </Tooltip>
      </div>
    );

    return {
      id: ID,
      title: Name,
      noApiSortFilter: [Name],
      cells: [
        {
          title: <Link to={`${paths['fleet-management']}/${ID}`}>{Name}</Link>,
        },
        {
          title: systems.length,
        },
        {
          title: ID === 1 ? image : 'Golden image',
        },
      ],
    };
  });

  return (
    <GeneralTable
      apiFilterSort={false}
      filters={filters}
      tableData={{
        count: data.length,
        isLoading,
        hasError: false,
      }}
      columnNames={columns}
      rows={buildRows}
      emptyState={{
        icon: 'module',
        title: 'Create a system group',
        body: 'Create system groups to help manage your devices more effectively',
        primaryAction: {
          text: 'Create group',
          click: () => setIsModalOpen(true),
        },
      }}
      emptyFilterState={{
        message: 'No matching groups found',
        body: 'To continue, edit your filter settings and try again',
      }}
      actionResolver={actionResolver}
      areActionsDisabled={() => false}
      defaultSort={{ index: 0, direction: 'desc' }}
      toolbarButtons={[
        {
          title: 'Create group',
          click: openModal,
        },
      ]}
    />
  );
};

GroupTable.propTypes = {
  data: PropTypes.array,
  openModal: PropTypes.func,
  setIsModalOpen: PropTypes.func,
  isLoading: PropTypes.bool,
  handleRenameModal: PropTypes.func,
};

export default GroupTable;
