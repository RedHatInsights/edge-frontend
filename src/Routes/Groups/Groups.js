import React from 'react';

const Groups = () => {
  return <div>Here will be groups!</div>;
};

export default Groups;
