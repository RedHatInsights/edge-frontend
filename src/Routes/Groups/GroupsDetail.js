import React from 'react';
import { useParams } from 'react-router-dom';

const Groups = () => {
    const { uuid } = useParams();
    return <div>Here will be group detail of - {uuid}!</div>;
};

export default Groups;
