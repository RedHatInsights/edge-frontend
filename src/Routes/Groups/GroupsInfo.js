import React, { useEffect, Fragment } from 'react';
import {
  Grid,
  GridItem,
  Card,
  CardBody,
  CardHeader,
  Bullseye,
  Spinner,
  Title,
  TextContent,
  TextList,
  TextListVariants,
  TextListItem,
  TextListItemVariants,
  Skeleton,
  Text,
  TextVariants,
  Level,
  LevelItem,
  CardFooter,
  Split,
  SplitItem,
} from '@patternfly/react-core';
import PropTypes from 'prop-types';
import { DateFormat } from '@redhat-cloud-services/frontend-components/DateFormat';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import {
  loadThreshold,
  loadDevicesInfo,
  loadCanariesInfo,
} from '../../store/actions';
import { StatusIcon } from '../../components';
import { ChartPie, ChartThemeColor } from '@patternfly/react-charts';
import { ArrowRightIcon } from '@patternfly/react-icons';

const GroupsInfo = ({ numberOfSystems }) => {
  const dispatch = useDispatch();
  const threshold = useSelector(
    ({ thresholdReducer }) => thresholdReducer?.threshold || {}
  );
  const isThresholdLoading = useSelector(
    ({ thresholdReducer }) => thresholdReducer?.isLoading
  );
  const isDevicesInfoLoading = useSelector(
    ({ devicesInfoReducer }) => devicesInfoReducer?.isLoading
  );
  const devicesInfo = useSelector(
    ({ devicesInfoReducer }) => devicesInfoReducer?.devicesInfo || {}
  );
  const canariesInfo = useSelector(({ canariesInfoReducer }) =>
    (canariesInfoReducer?.canariesInfo || []).slice(0, 4)
  );
  const isCanariesInfoLoading = useSelector(
    ({ canariesInfoReducer }) => canariesInfoReducer?.isLoading
  );
  useEffect(() => {
    dispatch(loadThreshold());
    dispatch(loadCanariesInfo());
  }, []);

  useEffect(() => {
    dispatch(loadDevicesInfo(numberOfSystems));
  }, [numberOfSystems]);

  return (
    <Grid hasGutter>
      <GridItem span={3}>
        <Card>
          <CardHeader>
            <Title headingLevel="h3">Health threshold</Title>
          </CardHeader>
          <CardBody>
            {isThresholdLoading === false ? (
              <ChartPie
                ariaDesc="Health threshold"
                ariaTitle="Health threshold"
                constrainToVisibleArea={true}
                legendOrientation="vertical"
                legendPosition="right"
                height={230}
                width={450}
                data={threshold.map((item) => ({
                  x: Object.keys(item)?.[0],
                  y: Object.values(item)?.[0],
                }))}
                legendData={threshold.map((item) => ({
                  name: `${Object.keys(item)?.[0]}: ${
                    Object.values(item)?.[0]
                  }`,
                }))}
                padding={{
                  bottom: 20,
                  left: 20,
                  right: 140, // Adjusted to accommodate legend
                  top: 20,
                }}
                labels={({ datum }) => `${datum.x}: ${datum.y}`}
                themeColor={ChartThemeColor.multiOrdered}
              />
            ) : (
              <Bullseye>
                <Spinner />
              </Bullseye>
            )}
          </CardBody>
        </Card>
      </GridItem>
      <GridItem span={3}>
        <Card>
          <CardHeader>
            <Title headingLevel="h3">Devices info</Title>
          </CardHeader>
          <CardBody>
            <TextContent>
              <TextList component={TextListVariants.dl}>
                <TextListItem component={TextListItemVariants.dt}>
                  {isDevicesInfoLoading === false ? (
                    <Text component={TextVariants.h1}>
                      <Link to="/groups">{devicesInfo?.requiredApproval}</Link>
                    </Text>
                  ) : (
                    <Skeleton />
                  )}
                </TextListItem>
                <TextListItem component={TextListItemVariants.dd}>
                  Required approvals
                </TextListItem>
                <TextListItem component={TextListItemVariants.dt}>
                  {isDevicesInfoLoading === false ? (
                    <Text component={TextVariants.h1}>
                      <Link to="/groups">{devicesInfo?.orphaned}</Link>
                    </Text>
                  ) : (
                    <Skeleton />
                  )}
                </TextListItem>
                <TextListItem component={TextListItemVariants.dd}>
                  Orphaned devices
                </TextListItem>
                <TextListItem component={TextListItemVariants.dt}>
                  {isDevicesInfoLoading === false ? (
                    <Text component={TextVariants.h1}>
                      <Link to="/groups">{devicesInfo?.delivering}</Link>
                    </Text>
                  ) : (
                    <Skeleton />
                  )}
                </TextListItem>
                <TextListItem component={TextListItemVariants.dd}>
                  Devices on the way
                </TextListItem>
              </TextList>
            </TextContent>
          </CardBody>
        </Card>
      </GridItem>
      <GridItem span={3}>
        <Card>
          <CardHeader>
            <Title headingLevel="h3">Last Canaries</Title>
          </CardHeader>
          <CardBody>
            <TextContent>
              <TextList component={TextListVariants.dl}>
                {isCanariesInfoLoading === true ? (
                  <Spinner />
                ) : (
                  canariesInfo.map(({ group, date, status }, key) => (
                    <Fragment key={group?.uuid || key}>
                      <TextListItem component={TextListItemVariants.dt}>
                        <Link to={`/groups/${group?.uuid}`}>{group?.name}</Link>
                      </TextListItem>
                      <TextListItem component={TextListItemVariants.dd}>
                        <Level>
                          <LevelItem>
                            <DateFormat date={date} />
                          </LevelItem>
                          <LevelItem>
                            <StatusIcon status={status} />
                          </LevelItem>
                        </Level>
                      </TextListItem>
                    </Fragment>
                  ))
                )}
              </TextList>
            </TextContent>
          </CardBody>
          <CardFooter>
            <Split>
              <SplitItem isFilled />
              <SplitItem>
                <Link to="/canaries">
                  See canaries <ArrowRightIcon />
                </Link>
              </SplitItem>
            </Split>
          </CardFooter>
        </Card>
      </GridItem>
    </Grid>
  );
};

GroupsInfo.propTypes = {
  numberOfSystems: PropTypes.number,
};

GroupsInfo.defaultProps = {
  numberOfSystems: 0,
};

export default GroupsInfo;
