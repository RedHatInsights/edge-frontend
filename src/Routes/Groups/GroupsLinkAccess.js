import React from 'react';
import {
  Button,
  EmptyState,
  EmptyStateIcon,
  EmptyStateBody,
  EmptyStateSecondaryActions,
  Icon,
  TextContent,
  Text,
  Title,
} from '@patternfly/react-core';
import { ExternalLinkAltIcon } from '@patternfly/react-icons';
import PageHeader from '@redhat-cloud-services/frontend-components/PageHeader';
import { useFeatureFlags } from '../../utils';
import { FEATURE_INVENTORY_WORKSPACES_RENAME } from '../../constants/features';

const GroupsLinkAccess = () => {
  const useWorkspacesRename = useFeatureFlags(
    FEATURE_INVENTORY_WORKSPACES_RENAME
  );

  return (
    <>
      <PageHeader>
        <TextContent>
          <Text component="h1" id="custom-repos-title">
            Groups
          </Text>
        </TextContent>
      </PageHeader>

      <EmptyState id="moved-state">
        <EmptyStateIcon id="external-link-icon" icon={ExternalLinkAltIcon} />
        <Title id="moved-state-title" headingLevel="h4" size="xl">
          Groups have moved
        </Title>
        <EmptyStateBody id="moved-state-body">
          You can now use your groups across the console. Access them on the
          Insights {useWorkspacesRename ? 'Workspaces' : 'Groups'} page.
        </EmptyStateBody>
        <Button
          id="moved-state-button"
          variant="primary"
          component="a"
          target="_blank"
          href={`https://${window.location.host}/insights/inventory/groups`}
        >
          Go to {useWorkspacesRename ? 'Workspaces' : 'Groups'}
        </Button>
        <EmptyStateSecondaryActions>
          <Button
            id="moved-state-link"
            variant="link"
            component="a"
            target="_blank"
            href="https://access.redhat.com/documentation/en-us/red_hat_insights/2023/html/viewing_and_managing_system_inventory/deploying-insights-with-rhca"
          >
            Learn more about {useWorkspacesRename ? 'Workspaces' : 'Groups'}
            <Icon
              style={{ paddingLeft: '1rem' }}
              iconSize="md"
              size="lg"
              isInline
            >
              <ExternalLinkAltIcon />
            </Icon>
          </Button>
        </EmptyStateSecondaryActions>
      </EmptyState>
    </>
  );
};

export default GroupsLinkAccess;
