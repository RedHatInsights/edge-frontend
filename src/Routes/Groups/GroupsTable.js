import React, { useState } from 'react';
import { useSelector } from 'react-redux';
import {
  Table,
  TableHeader,
  TableBody,
  sortable,
  cellWidth,
} from '@patternfly/react-table';
import { DateFormat } from '@redhat-cloud-services/frontend-components';
import { StatusIcon, SecureIcon } from '../../components';
import { Link } from 'react-router-dom';

const columns = [
  {
    title: 'Name of group',
    transforms: [sortable, cellWidth('max')],
  },
  {
    title: 'Number of devices',
    transforms: [sortable],
  },
  {
    title: 'Secure',
    transforms: [sortable, cellWidth(10)],
  },
  {
    title: 'Last seen',
    transforms: [sortable, cellWidth(15)],
  },
  {
    title: 'Status',
    transforms: [sortable, cellWidth(10)],
  },
];

const GroupsTable = () => {
  const [sortBy, setSortBy] = useState({});
  const groups = useSelector(({ groupsReducer }) => groupsReducer?.groups);
  return (
    <Table
      aria-label="Groups table"
      actions={[
        {
          title: 'Adopt',
          onClick: console.log,
        },
        {
          title: 'Reject',
          onClick: console.log,
        },
      ]}
      cells={columns}
      sortBy={sortBy}
      onSort={(_e, index, direction) =>
        setSortBy(() => ({
          index,
          direction,
        }))
      }
      rows={groups.map((group) => ({
        cells: [
          {
            title: <Link to={`/groups/${group?.uuid}`}>{group?.name}</Link>,
          },
          group?.sensors,
          {
            title: <SecureIcon isSecure={group?.is_secure} />,
          },
          {
            title: <DateFormat date={group?.last_seen} />,
          },
          {
            title: <StatusIcon status={group?.status} />,
          },
        ],
      }))}
    >
      <TableHeader />
      <TableBody />
    </Table>
  );
};

export default GroupsTable;
