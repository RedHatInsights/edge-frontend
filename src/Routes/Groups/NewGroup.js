import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { FormRenderer } from '@data-driven-forms/react-form-renderer';
import {
  FormTemplate as Pf4FormTemplate,
  componentMapper,
} from '@data-driven-forms/pf4-component-mapper';
import schema from './newGroupSchema';
import { FormInventoryTable, INVENTORY_TABLE } from '../../components/form';
import ReducerRegistry from '@redhat-cloud-services/frontend-components-utilities/files/esm/ReducerRegistry';
import { InventoryContext } from '../../components/form/constants';
import promiseMiddleware from 'redux-promise-middleware';

const FormTemplate = (props) => (
  <Pf4FormTemplate {...props} showFormControls={false} />
);

const NewGroup = ({ onAction }) => {
  const [registry, setRegistry] = useState();
  useEffect(() => {
    setRegistry(
      new ReducerRegistry(
        {
          selected: new Map(),
        },
        [promiseMiddleware]
      )
    );
  }, []);

  return (
    <InventoryContext.Provider value={{ registry }}>
      <FormRenderer
        schema={schema}
        FormTemplate={FormTemplate}
        componentMapper={{
          ...componentMapper,
          [INVENTORY_TABLE]: FormInventoryTable,
        }}
        onSubmit={(values) => {
          const { entities } = registry.store.getState();
          onAction(true, {
            ...values,
            selected: Array.from(entities?.selected?.keys() || []),
          });
        }}
        onCancel={(...args) => onAction(false, ...args)}
      />
    </InventoryContext.Provider>
  );
};

NewGroup.propTypes = {
  isOpened: PropTypes.bool,
  onAction: PropTypes.func,
};

NewGroup.defaultProps = {
  onAction: () => {},
};

export default NewGroup;
