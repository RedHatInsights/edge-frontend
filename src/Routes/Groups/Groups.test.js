import React from 'react';
import Groups from './Groups';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import { MemoryRouter, Route } from 'react-router-dom';

describe('Groups', () => {
  const mockStore = configureStore();
  it('should render correctly', () => {
    const { container } = render(
      <Provider store={mockStore({})}>
        <MemoryRouter initialEntries={['/groups']}>
          <Route path="/groups">
            <Groups />
          </Route>
        </MemoryRouter>
      </Provider>
    );
    expect(container.querySelector('div')).toMatchSnapshot();
  });
});
