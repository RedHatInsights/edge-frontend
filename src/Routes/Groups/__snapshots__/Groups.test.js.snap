// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Groups should render correctly 1`] = `
<div
  class="pf-c-toolbar"
  data-ouia-component-id="OUIA-Generated-Toolbar-1"
  data-ouia-component-type="PF4/Toolbar"
  data-ouia-safe="true"
  data-testid="toolbar-header-testid"
  id="toolbar-header"
  style="padding: 0px;"
>
  <div
    class="pf-c-toolbar__content"
  >
    <div
      class="pf-c-toolbar__content-section"
    >
      <div
        class="pf-c-toolbar__item"
        data-testid="filter-input-testid"
      >
        <div
          class="pf-c-input-group"
        >
          <div
            class="pf-c-text-input-group"
            id="textInput1"
            name="textInput1"
          >
            <div
              class="pf-c-text-input-group__main pf-m-icon"
            >
              <span
                class="pf-c-text-input-group__text"
              >
                <span
                  class="pf-c-text-input-group__icon"
                >
                  <svg
                    aria-hidden="true"
                    fill="currentColor"
                    height="1em"
                    role="img"
                    style="vertical-align: -0.125em;"
                    viewBox="0 0 512 512"
                    width="1em"
                  >
                    <path
                      d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
                    />
                  </svg>
                </span>
                <input
                  aria-label="Select input for name"
                  class="pf-c-text-input-group__text-input"
                  placeholder="Filter by name"
                  type="search"
                  value=""
                />
              </span>
            </div>
            
          </div>
        </div>
      </div>
      <div
        class="pf-c-toolbar__item"
      >
        <button
          aria-disabled="false"
          class="pf-c-button pf-m-primary"
          data-ouia-component-id="OUIA-Generated-Button-primary-1"
          data-ouia-component-type="PF4/Button"
          data-ouia-safe="true"
          type="button"
        >
          Create group
        </button>
      </div>
      <div
        align="[object Object]"
        class="pf-c-toolbar__item pf-m-pagination"
      >
        <div
          class="pf-c-skeleton"
          style="--pf-c-skeleton--Width: 200px;"
        >
          <span
            class="pf-u-screen-reader"
          />
        </div>
      </div>
    </div>
    <div
      class="pf-c-toolbar__expandable-content"
      id="toolbar-header-expandable-content-0"
    >
      <div
        class="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    class="pf-c-toolbar__content"
  >
    <div
      class="pf-c-toolbar__content-section"
    >
      <div
        class="pf-c-toolbar__item pf-m-chip-group pf-m-spacer-none"
      />
    </div>
    <div
      class="pf-c-toolbar__expandable-content"
      id="toolbar-header-expandable-content-1"
    >
      <div
        class="pf-c-toolbar__group"
      />
    </div>
  </div>
  <div
    class="pf-c-toolbar__content pf-m-hidden"
    hidden=""
  >
    <div
      class="pf-c-toolbar__group"
    />
  </div>
</div>
`;
